### DISCUSSION TITLE
Should record IDs be GUIDs?

### DISCUSSION THREAD

**Discussion by msheller** [mlcommons/medperf#242] [Ideas]
Currently, a registered dataset is given an auto-incremented ID, such that the ID is unique only in server context. This means that our filesystem could have multiple entries for the exact same data if a user were to register a dataset using the same prep logic on two different servers.

Alternatively, dataset IDs could be GUIDs, such that we could potentially submit the same registration to multiple servers and keep the datasets in a server-agnostic folder. To aid this, the client could do the same generated UID checks the server does when the server tries to prevent duplicate entries. This way, the client can check for this problem in cases split across servers. (Does the client perhaps already do this to prevent erroneous computation if the client tries to prepare the same dataset the same way multiple times?

**Comment by hasan7n** [comment]
converted to an issue
